tracer/src/Datadog.Trace/AppSec/Transports/SecurityTransport.cs

```diff
@@ -0,0 +1,104 @@
+// <copyright file="SecurityTransport.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.AppSec.Waf;
+using Datadog.Trace.AppSec.Waf.ReturnTypes.Managed;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.Serilog.Events;
+
+namespace Datadog.Trace.AppSec.Transports;
+
+/// <summary>
+/// Bridge class between security components and http transport classes, that calls security and is responsible for reporting
+/// </summary>
+internal partial class SecurityTransport : ISecurityTransport
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<SecurityTransport>();
+    private readonly Security _security;
+    private readonly Span _localRootSpan;
+    private readonly HttpTransportBase _httpTransport;
+
+    private SecurityTransport(Span span, HttpTransportBase httpTransport, Security security)
+    {
+        _security = security;
+        _localRootSpan = TryGetRoot(span);
+        _httpTransport = httpTransport;
+    }
+
+    public bool Blocked => _httpTransport.Blocked;
+
+    public void MarkBlocked() => _httpTransport.MarkBlocked();
+
+    private static void LogMatchesIfDebugEnabled(string result, bool blocked)
+    {
+        if (Log.IsEnabled(LogEventLevel.Debug))
+        {
+            var results = JsonConvert.DeserializeObject<WafMatch[]>(result);
+            for (var i = 0; i < results?.Length; i++)
+            {
+                var match = results[i];
+                Log.Debug(blocked ? "DDAS-0012-02: Blocking current transaction (rule: {RuleId})" : "DDAS-0012-01: Detecting an attack from rule {RuleId}", match.Rule);
+            }
+        }
+    }
+
+    public IResult ShouldBlock()
```

Comment:
nit: not a fan of this name as calling `transport.ShouldBlock()` seems like you're telling it that it should block, not checking _if_ it should block. 

It's a bit wordy, but how about something like one of these?

- `CheckWhetherToBlock()`
- `CheckBlockStatus()`
- `CheckIsBlockingRequired()`
- `IsBlockingRequired()`

something like that?

Reply:
fixed in 60213612bd9dc48fe4f254b95d896f2a95e759d4